planner/core/task.go

```diff
@@ -782,7 +782,12 @@ func (p *PhysicalHashJoin) attach2TaskForMpp(tasks ...task) task {
 	lCost := lTask.cost()
 	rCost := rTask.cost()
 
-	outerTask := tasks[1-p.InnerChildIdx].(*mppTask)
+	// can not use the task from tasks because it maybe updated.
+	outerTaskIndex := 1 - p.InnerChildIdx
```

Comment:
does this happen in `convertPartitionKeysIfNeed`?

Reply:
No, `convertPartitionKeysIfNeed` returns updated mpp task directly.